build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            if (!this.versionService.IsVersionWithinRange(
+                releaseVersion,
+                minVersion: this.dotNetMinReleaseVersion,
+                maxVersion: this.dotNetMaxReleaseVersion))
+            {
+                return true;
+            }
+
+            if (this.oryxDotNetSdkVersions.Contains(sdkVersion))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private DotNetVersion GetDotNetVersion(Release release, string versionType)
+        {
+            string version;
+            string sha;
+            switch (versionType)
+            {
+                case DotNetConstants.SdkName:
+                    sha = this.GetSha(release.Sdk.Files);
+                    version = release.Sdk.Version;
+                    break;
+                case DotNetConstants.DotNetCoreName:
+                    sha = this.GetSha(release.Runtime.Files);
+                    version = release.Runtime.Version;
+                    break;
+                case DotNetConstants.DotNetAspCoreName:
+                    sha = this.GetSha(release.AspNetCoreRuntime.Files);
+                    version = release.AspNetCoreRuntime.Version;
+                    break;
+                default:
+                    throw new InvalidDataException($"Invalid DotNet version type: {versionType}");
+            }
+
+            return new DotNetVersion
+            {
+                Version = version,
+                Sha = sha,
+                VersionType = versionType,
+            };
+        }
+
+        private void UpdateOryxConstantsForNewVersions(List<DotNetVersion> dotNetVersions, List<ConstantsYamlFile> yamlConstants)
+        {
+            Dictionary<string, ConstantsYamlFile> dotnetYamlConstants = this.GetYamlDotNetConstants(yamlConstants);
+
+            // update dotnetcore sdks and runtimes
+            foreach (var dotNetVersion in dotNetVersions)
+            {
+                string version = dotNetVersion.Version;
+                string sha = dotNetVersion.Sha;
+                string versionType = dotNetVersion.VersionType;
+                string dotNetConstantKey = this.GenerateDotNetConstantKey(dotNetVersion);
+                Console.WriteLine($"[UpdateConstants] version: {version} versionType: {versionType} sha: {sha} dotNetConstantKey: {dotNetConstantKey}");
+
+                if (versionType.Equals(DotNetConstants.SdkName))
+                {
+                    ConstantsYamlFile dotNetYamlConstant = dotnetYamlConstants[DotNetConstants.DotNetSdkKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // update versionsToBuild.txt
+                    this.fileService.UpdateVersionsToBuildTxt(
+                        DotNetConstants.DotNetName,
+                        $"\n{dotNetVersion.Version}, {dotNetVersion.Sha},");
```

Comment:
nit: do we need to provide the newline character as a prefix to the second argument in `UpdateVersionsToBuildTxt` or can we automatically set it inside of the method when the argument value is used?

Reply:
Doesn't seem to need it. But I've added an `Environment.NewLine` after the `line` that is provided in the `FileService`